event/batch.go

```diff
@@ -64,9 +65,15 @@ func (batch *Batch) IsEmpty() bool {
 	return len(batch.events) == 0
 }
 
-// Commit is called when the batch has been processed.
+// Commit is called when the batch has been processed. The last message has been released already, so at this point we just need to commit
 func (batch *Batch) Commit() {
-	batch.lastMessageInBatch.Commit()
+	for i, msg := range batch.messages {
+		if i < len(batch.messages)-1 {
+			msg.Mark()
```

Comment:
The `else` block isn't necessary here, you could achieve the same with:
```
if i < len(batch.messages)-1 {
    msg.Mark()
    continue
}
msg.Commit()
```
Functionally the same but a bit less code. I won't hold up this change because of this.